Lib/test/test_pickle.py

```diff
@@ -112,6 +113,66 @@ class PyIdPersPicklerTests(AbstractIdentityPersistentPicklerTests,
     pickler = pickle._Pickler
     unpickler = pickle._Unpickler
 
+    @support.cpython_only
+    def test_pickler_reference_cycle(self):
+        def check(Pickler):
+            for bin in 0, 1:
```

Comment:
Hmm, what is `bin`? Do you mean `protocol`?

Reply:
A flag text/binary. Later it became a protocol, but the attribute `bin` still is used in sources (it is equal `protocol != 0`). Here it is enough to test only one binary protocol.

Reply:
Yes... It's still a bit cryptic though. And it doesn't cost anything to loop over all protocols, right? :-)

Reply:
Done.